tests/common/include/tests/common/tools/create_toy_geometry.hpp

```diff
@@ -830,23 +873,30 @@ auto create_toy_geometry(vecmem::memory_resource &resource,
     // Don't create modules in gap volume
     struct empty_vol_factory {
         void operator()(
-            typename detector_t::context & /*ctx*/, dindex /*volume_id*/,
+            typename detector_t::context & /*ctx*/,
+            typename detector_t::volume_type & /*volume*/,
             typename detector_t::surface_filling_container & /*surfaces*/,
+            typename detector_t::surfaces_regular_circular_grid
```

Comment:
Hm, just a thought here: A surface factory is meant to create surfaces, not to additionally define a grid. Designwise, I think, this mixes things up too much...
This of course translates to the surfaces creation functions, as well. Do you have a specific reason to do it this way?

Reply:
I don't have a specific reason. What would be the best way?

Reply:
Hmm I think I can figure out this

Reply:
I separated the functions for surface creation and grid creation. Probably it is better to read

Reply:
Ok, let's leave it like this